### PR TITLE
Feature - Add DomainCreatePage + Change other "modal" page layouts

### PR DESCRIPTION
**Pull request by krishan711**
Screen Shot 2021-01-20 at 18 38 35
Screen Shot 2021-01-20 at 18 38 42

### CODE REVIEW

File: app/src/App.tsx [hunk 1]
```diff
@@ -70,7 +70,7 @@ export const App = (): React.ReactElement => {
         <Route path='/register' page={RegistrationPage} />
         <Route path='/forgot-password' page={ForgotPasswordPage} />
         <Route path='/domains' page={DomainsPage} />
-        <Route path='/domains/create' page={CreateDomainPage} />
+        <Route path='/domains/create' page={DomainCreatePage} />
```
Comment (rikhilrai) [inline app/src/App.tsx:73]: why this renaming?

File: app/src/pages/domainCreatePage/domainCreatePage.tsx [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+      setUrlError('Enter an url address');
+      hasErrors = true;
+    }
+    if (!hasErrors) {
+      setIsLoading(true);
+      try {
+        await asyncSleep(1000);
+        history.navigate('/domains');
+      } catch (error) {
+        // TODO(krishan711): error to be rendered when login fails
```
Comment (rikhilrai) [inline app/src/pages/domainCreatePage/domainCreatePage.tsx]: nit
```suggestion
        // TODO(krishan711): error to be rendered when domain creation fails
```

File: app/src/pages/forgotPasswordPage/forgotPasswordPage.tsx [hunk 3]
```diff
@@ -51,20 +51,23 @@ export const ForgotPasswordPage = (props: IForgotPasswordPageProps): React.React
...
+              messageText={emailError || undefined}
+              placeholderText='Email Address'
+              inputType={InputType.Email}
+              value={email}
+              onValueChanged={onEmailTyped}
+            />
+            <Spacing variant={PaddingSize.Wide} />
+            <Stack direction={Direction.Horizontal} shouldAddGutters={true} childAlignment={Alignment.Center}>
+              <Stack.Item growthFactor={1} shrinkFactor={1} />
+              <Button buttonType='submit' variant='primary' text='Email me a recovery link'/>
```
Comment (rikhilrai) [inline app/src/pages/forgotPasswordPage/forgotPasswordPage.tsx]:
```suggestion
              <Button buttonType='submit' variant='primary' text='Email me a recovery link' />
```

File: app/src/App.tsx (hunk 1 above)
Comment (krishan711) [reply app/src/App.tsx:73]: so it follows the url more closely and also the pages appear together in the folders

File: app/src/pages/loginPage/loginPage.tsx [hunk 4]
```diff
@@ -70,44 +70,47 @@ export const LoginPage = (props: ILoginPageProps): React.ReactElement => {
...
-                placeholderText='Password'
-                inputType={InputType.Password}
-                value={password}
-                onValueChanged={onPasswordTyped}
-              />
-              <Spacing variant={PaddingSize.Narrow} />
-              <Stack direction={Direction.Horizontal} shouldAddGutters={true} childAlignment={Alignment.Center}>
-                <Stack.Item growthFactor={1} shrinkFactor={1} />
-                <Link shouldOpenSameTab={true} text='Forgot Password?' target='/forgot-password' />
+      <ResponsiveContainingView sizeResponsive={{ base: 12, small: 8, medium: 6, large: 5 }}>
```
Comment (rikhilrai) [inline app/src/pages/loginPage/loginPage.tsx:73]: nice 👍

File: app/src/App.tsx (hunk 1 above)
Comment (krishan711) [reply app/src/App.tsx:73]: didnt actualyl realise a page existed for it already lol, happy to change back if you think its better the other way?

File: app/src/pages/loginPage/loginPage.tsx (hunk 4 above)
Comment (krishan711) [reply app/src/pages/loginPage/loginPage.tsx:73]: ive got a task to fix the `ResponsiveContainingView` having to be emedded in the `ContainingView` in ui-react. will update here once thats fixed

**Comment by krishan711** [comment]
also worth noting the vertical alignment isnt working for a wierd reason but have a task to fix

File: app/src/pages/forgotPasswordPage/forgotPasswordPage.tsx [hunk 5]
```diff
@@ -74,19 +77,24 @@ export const ForgotPasswordPage = (props: IForgotPasswordPageProps): React.React
...
     );
   };
 
   return (
     <ContainingView>
-      <Spacing variant={PaddingSize.Wide} />
-      <Stack isFullHeight={true} isFullWidth={true} shouldAddGutters={true} direction={Direction.Vertical} childAlignment={Alignment.Center}>
-        {hasRecoverPasswordFormBeenSubmitted ? renderEmailSentForm() : renderRecoverPasswordForm()}
-      </Stack>
+      <ResponsiveContainingView sizeResponsive={{ base: 12, small: 8, medium: 6, large: 5 }}>
```
Comment (rikhilrai) [inline app/src/pages/forgotPasswordPage/forgotPasswordPage.tsx:91]: what does base mean, what do these numbers mean?

Comment (krishan711) [reply app/src/pages/forgotPasswordPage/forgotPasswordPage.tsx:91]: its for responsiveness. base is i think <560px (or something close) small is 560-760 (or something close) and so on. it just makes sure the box stays looking nice in multiple sizes (copied from everypage console)

File: app/src/pages/loginPage/loginPage.tsx [hunk 6]
```diff
@@ -70,44 +70,47 @@ export const LoginPage = (props: ILoginPageProps): React.ReactElement => {
...
+        <Stack direction={Direction.Vertical} paddingVertical={PaddingSize.Wide2} isFullHeight={true}>
```
Comment (rikhilrai) [inline app/src/pages/loginPage/loginPage.tsx:74]: i don't fully understand how to structure these yet. maybe chat about this over the weekend lol

File: app/src/pages/domainCreatePage/domainCreatePage.tsx [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+        history.navigate('/domains');
+      } catch (error) {
+        // TODO(krishan711): error to be rendered when domain creation fails
+        console.error('error', error);
+      }
+    }
+    setIsLoading(false);
+  };
+
+  const onUrlChanged = (typedEmail: string): void => {
```
Comment (rikhilrai) [inline app/src/pages/domainCreatePage/domainCreatePage.tsx]:
```suggestion
  const onUrlChanged = (typedUrl: string): void => {
```

File: app/src/pages/domainCreatePage/domainCreatePage.tsx [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+    if (urlError) {
+      setUrlError(null);
+    }
+    setUrl(typedEmail);
```
Comment (rikhilrai) [inline app/src/pages/domainCreatePage/domainCreatePage.tsx]:
```suggestion
    setUrl(typedUrl);
```

File: app/src/pages/loginPage/loginPage.tsx (hunk 6 above)
Comment (krishan711) [reply app/src/pages/loginPage/loginPage.tsx:74]: ye that would be great, need to make it easier to use!

File: app/src/App.tsx (hunk 1 above)
Comment (rikhilrai) [reply app/src/App.tsx:73]: nah, this is cool 👍

Comment (rikhilrai) [reply app/src/App.tsx:73]: the other way reads better but this follows nicely from the path

Comment (krishan711) [reply app/src/App.tsx:73]: ah next step is autogeneration 😛

File: app/src/App.tsx [hunk 9]
```diff
@@ -70,7 +70,7 @@ export const App = (): React.ReactElement => {
...
         <Route path='/domains/:domainId' page={DomainPage} />
         <Route path='/domains/:domainId/links/:linkId' page={LinkPage} />
         <Route path='/domains/:domainId/links/create' page={CreateLinkPage} />
```
Comment (rikhilrai) [inline app/src/App.tsx:76]: actually on second thoughts, we should define a convention and stick to it. the alternative for this is DomainLinkCreatePage

**Review by rikhilrai — APPROVED** [review]

File: app/src/App.tsx (hunk 9 above)
Comment (krishan711) [reply app/src/App.tsx:76]: yep will update in that PR (hopefully later today) 👍

**Pull request merged**